packages/vitest/src/integrations/env/happy-dom.ts

```diff
@@ -5,9 +5,9 @@ import { populateGlobal } from './utils'
 export default <Environment>({
   name: 'happy-dom',
   transformMode: 'web',
-  async setupVM() {
+  async setupVM({ 'happy-dom': happyDom = {} }) {
     const { Window } = await importModule('happy-dom') as typeof import('happy-dom')
-    const win = new Window() as any
+    const win = new Window(happyDom) as any
```

Comment:
can you include default url?:
```ts
    const {
      innerWidth,
      innerHeight,
      url = 'http://localhost:3000',
      settings,
    } = happyDom
    const win = new (GlobalWindow || Window)({
      innerWidth,
      innerHeight,
      url,
      settings,
    })
```

Reply:
The issue for the default URL is #3905 (there is also a PR up that most likely will conflict when either is merged).

Reply:
uhmm, check the title, it is about provide happy-dom env, maybe it would be better to include my suggestion here and remove default rul form your PR.

Reply:
If this PR is approved, it is of course better to include both. Another option would be to merge #3907 first, as it is smaller and have a low risk.

Reply:
Wouldn't that be a breaking change? I mean, if by default everyone is using the `about:blank` and suddenly it flips to `http://localhost:3000`, wouldn't break things? 🤔

Reply:
I think that the scenarios where it could be breaking are very uncommon and most likely it is not breaking for anyone, but it is a good point.

The only scenario I can think of for now is if someone directly test against the current page URL (location.*) somehow.

Reply:
I also want to add that I think it is expected that JSDOM and Happy DOM are using the same URL when switching between environments.

Reply:
I've added in this PR the fallback url